### PR TITLE
feat: Add --strip-all to optimized linker flags

### PR DESCRIPTION
**Pull request by matte1**

### CODE REVIEW

**Comment by agoessling** [comment]
Is it possible make this a separate feature?  I don't think this would match the behavior of the `opt` config of the built-in toolchains.

**Comment by matte1** [comment]
For our use case we would like this tied to the opt feature specifically but it would be unexpected behavior for standard users. I made a separate feature flag to allow enabling the strip all feature.

If there was another way to associate optimized builds with --strip-all I would be interested in using that

File: toolchains/toolchains.bzl [hunk 1]
```diff
@@ -224,6 +224,13 @@ def _impl_cc_arm_toolchain_config(ctx):
                 ])],
                 with_features = [with_feature_set(features = ["opt"])],
             ),
+            # flag_set(
```
Comment (agoessling) [inline toolchains/toolchains.bzl]: Remove

Comment (matte1) [reply toolchains/toolchains.bzl]: Done

**Comment by agoessling** [comment]
Have you confirmed this indeed does not strip unless the config is specified on the command line?

**Comment by matte1** [comment]
`main: statically linked, stripped`
with --features strip_all and -c opt

`main: statically linked, not stripped`
without the feature and -c opt

**Comment by matte1** [comment]
Using file

**Review by agoessling — APPROVED** [review]

**Pull request merged**